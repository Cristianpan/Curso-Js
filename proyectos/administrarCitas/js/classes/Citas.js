export default class Citas {
  constructor() {
    this.citas = [];
  }

  agregarCitas(citas) {
    this.citas = [...this.citas, ...citas];
  }

  agregarCita(cita) {
    this.citas = [...this.citas, cita];

    console.log(this.citas);
  }

  eliminarCita(id) {
    this.citas = this.citas.filter((cita) => cita.id !== id);
  }

  modificarCita(cita) {
    this.citas = this.citas.map((elemento) =>
      elemento.id === cita.id ? cita : elemento
    );
  }
}
