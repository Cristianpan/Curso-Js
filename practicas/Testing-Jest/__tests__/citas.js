import Citas from "../js/classes/Citas.js";

describe("Probar la clase de Citas", () => {
  const citas = new Citas();
  const id = Date.now();

  test("Agregar una nueva cita", () => {
    const citaObj = {
      id,
      mascota: "Michi",
      propietario: "Pedro",
      telefono: "9993987865",
      fecha: "10-02-2024",
      hora: "11:50",
      sintomas: "Solo duerme",
    };
    const citaObj2 = {
      id: Date.now(), 
      mascota: "Sorry",
      propietario: "Pedro",
      telefono: "9993987865",
      fecha: "10-02-2024",
      hora: "11:50",
      sintomas: "Solo duerme",
    };

    citas.agregarCita(citaObj);
    citas.agregarCita(citaObj2);

    expect(citas).toMatchSnapshot();
  });

  test("Actualizar una nueva cita", () => {
    const citaObj = {
      id,
      mascota: "Nuevo nombre",
      propietario: "Pedro",
      telefono: "9993987865",
      fecha: "10-02-2024",
      hora: "11:50",
      sintomas: "Solo duerme",
    };

    citas.editarCita(citaObj);

    expect(citas).toMatchSnapshot();
  });

  test("Eliminar cita", ()=> {
    citas.eliminarCita(id); 

    expect(citas).toMatchSnapshot(); 
  }); 
});
