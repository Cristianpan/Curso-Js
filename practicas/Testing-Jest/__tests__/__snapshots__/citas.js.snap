// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Actualizar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-02-2024",
      "hora": "11:50",
      "id": 1709652862339,
      "mascota": "Nuevo nombre",
      "propietario": "Pedro",
      "sintomas": "Solo duerme",
      "telefono": "9993987865",
    },
    {
      "fecha": "10-02-2024",
      "hora": "11:50",
      "id": 1709652862347,
      "mascota": "Sorry",
      "propietario": "Pedro",
      "sintomas": "Solo duerme",
      "telefono": "9993987865",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-02-2024",
      "hora": "11:50",
      "id": 1709652862339,
      "mascota": "Michi",
      "propietario": "Pedro",
      "sintomas": "Solo duerme",
      "telefono": "9993987865",
    },
    {
      "fecha": "10-02-2024",
      "hora": "11:50",
      "id": 1709652862347,
      "mascota": "Sorry",
      "propietario": "Pedro",
      "sintomas": "Solo duerme",
      "telefono": "9993987865",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-02-2024",
      "hora": "11:50",
      "id": 1709652862347,
      "mascota": "Sorry",
      "propietario": "Pedro",
      "sintomas": "Solo duerme",
      "telefono": "9993987865",
    },
  ],
}
`;
